app/Http/Controllers/ItemController.php

```diff
@@ -445,13 +453,40 @@ public function submitMassInput($guildSlug) {
 
         AuditLog::insert($audits);
 
-        // For each item added, attempt to delete a matching item from the character's wishlist
+        // For each item added, attempt to delete a matching item from the character's wishlist and their prios
         foreach ($detachRows as $detachRow) {
+            // Remove from wishlist
             DB::table('character_items')->where([
                 'item_id'      => $detachRow['item_id'],
                 'character_id' => $detachRow['character_id'],
-                'type'         => $detachRow['type'],
-            ])->limit(1)->delete();
+                'type'         => Item::TYPE_WISHLIST,
+            ])->limit(1)->orderBy('order')->delete();
+
+            // Find prio for this item
+            $row = DB::table('character_items')->where([
+                'item_id'      => $detachRow['item_id'],
+                'character_id' => $detachRow['character_id'],
+                'type'         => Item::TYPE_PRIO,
+            ])->orderBy('order')->first();
+
+            // Does a prio exist
+            if ($row) {
+                // Delete the first one we find
+                DB::table('character_items')->where([
+                    'item_id'      => $detachRow['item_id'],
+                    'character_id' => $detachRow['character_id'],
+                    'type'         => Item::TYPE_PRIO,
+                ])->orderBy('order')->limit(1)->delete();
+
+                // Now correct the ordder on the remaning prios for that item in that raid
+                DB::table('character_items')->where([
+                        'item_id' => $row->item_id,
+                        'raid_id' => $row->raid_id,
+                        'type'    => Item::TYPE_PRIO,
+                    ])
+                    ->where('order', '>', $row->order)
+                    ->update(['order' => DB::raw('`order` - 1')]);
+            }

```

Comment:
@Pyrodius0 I added the code to delete **one** prio item when someone gets an item assigned to them. It will also correct the ordering of the other people prio'd that item for that raid.

(eg. Bill is prio 1, Bob is prio 2, Jill is prio 3, Jane is prio 4. Bob gets the item and has his prio removed. Jill is moved to prio 2. Jane is moved to prio 3.)

Could you please test this functionality when you get a chance? It's a destructive change that cannot be undone, and I'm doing this at 4-5am because I couldn't sleep.

Thank you!

(also the other requested changes were added)

Reply:
So testing out this functionality, I had some weird behavior.. I was attempting to assign it to a player that had the item priority in both raids.. I did it twice and it never assigned anything. 

I went to do it and take screen shots.. and it worked. 

I might have had some bad cache or its an actual issue.

Oh, its a behavior of the assign loot page. 

If you copy and paste the name of the item into the item field and dont click it, it still lets you submit the form and looks like you assigned an item. 

This looks good


The note at the top of the priority page, should we put  in something that says "When someone receives an item, we'll attempt to automatically remove it from their prios. If the user has the same item prioritized to them in multiple raids, we will remove it from the first one we find."

Reply:
[Change the item input styles when assigning loot.](https://github.com/thatsmybis/thatsmybis/issues/32) will help with the UI issue on the input page.

I added an audit log for when we pop the item off the prio list so that users can review what happened.
![image](https://user-images.githubusercontent.com/3407841/90829694-16773f00-e2f5-11ea-9a4d-ff16957bfb6f.png)

It may also be worth noting that we're going to remove the highest prio the user has for that item. If one prio has them at #1 and another has them at #3, we remove #1. Hopefully this means we're removing the one they were actually in line for, unless they had multiples at the same rank.